### PR TITLE
New Feature: Managing Host Packages

### PR DESCRIPTION
**Pull request by mjivraja**
In the Project Web UI, now end-users can easily manage the packages required for hosts. They can install, upgrade or remove the host packages from the Packages tab itself. In addition to this, they can upgrade or remove multiple packages via a single click. The end-users will also have options to choose from the available upgrade method as well while upgrading the packages. The packages page is paginated for better navigation between pages.

https://www.youtube.com/watch?v=ykQFM7aFczY&t=1471s

* [X] I am familiar with the [contributing](https://github.com/theforeman/foreman-documentation/blob/master/CONTRIBUTING.md) guidelines.

Please cherry-pick my commits into:

* [X] Foreman 3.5/Katello 4.7 (planned Satellite 6.13)
* [X] Foreman 3.4/Katello 4.6
* [X] Foreman 3.3/Katello 4.5 (Satellite 6.12)
* [X] Foreman 3.2/Katello 4.4
* [X] Foreman 3.1/Katello 4.3 (Satellite 6.11, orcharhino 6.1+)
* For Foreman 3.0 or older, please create a separate PR.
* We do not accept PRs for Foreman 2.3 or older.

### CODE REVIEW

File: guides/common/modules/con_managing-host-packages.adoc [hunk 1]
```diff
@@ -0,0 +1,9 @@
+[id="Managing-Host-Packages_{context}"]
+= Managing Host Packages
+
+You can manage the packages of hosts even if the subscriptions are not attached to access repositories.
+Navigate to *Hosts > All Hosts* and select any Host.
```
Comment (maximiliankolb) [inline guides/common/modules/con_managing-host-packages.adoc:5]: This reads like a procedure.

Comment (mjivraja) [reply guides/common/modules/con_managing-host-packages.adoc]: Addressed :+1:

But confused if I should rename the file to `proc` from `con`.

Comment (maximiliankolb) [reply guides/common/modules/con_managing-host-packages.adoc]: Yes, please. I am unsure it it makes sense to still split this into two files: "con_" and "proc_". Maybe someone else can weigh in? @bangelic

Comment (bangelic) [reply guides/common/modules/con_managing-host-packages.adoc]: This reads  similar to proc_removing-host-packages so this should probably be renamed to proc_managing_host_packages.adoc

Comment (mjivraja) [reply guides/common/modules/con_managing-host-packages.adoc]: Alright. Thank you @bangelic 
Addressed :+1:

File: guides/common/modules/proc_managing-host-packages.adoc [hunk 2]
```diff
@@ -0,0 +1,12 @@
+[id="Managing-Host-Packages_{context}"]
+= Managing Host Packages
+
+You can manage the packages of hosts even if the subscriptions are not attached to access repositories.
+
+.Procedure
+. Navigate to *Hosts > All Hosts* and select any Host.
+. Click on the *Content* tab and go to *Packages* where you can see the list of all the available packages with their status.
+
+The Packages pages are paginated, use the arrows to navigate through the pages.
```
Comment (maximiliankolb) [inline guides/common/modules/proc_managing-host-packages.adoc:10]: A lot of pages are paginated; I have never seen this note before. I am unsure if we want to add this to our docs. cc @Lennonka What do you think?

File: guides/common/modules/proc_managing-host-packages.adoc [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+
+You can *Install packages* available for that particular host by clicking the vertical ellipsis on the top.
```
Comment (maximiliankolb) [inline guides/common/modules/proc_managing-host-packages.adoc:12]: Is this part of the procedure? Should it be part of the list? Maybe reword to ". Click ..."

File: guides/common/modules/proc_removing-host-packages.adoc [hunk 4]
```diff
@@ -0,0 +1,9 @@
+[id="Removing-Host-Packages_{context}"]
+= Removing Host Packages
+
+To remove a particular package, click *Remove* from the vertical ellipsis of individual packages.
```
Comment (maximiliankolb) [inline guides/common/modules/proc_removing-host-packages.adoc:4]: This file is missing a short introduction like `You can use {Project} to remove packages from managed hosts.` and `.Procedure` with a ordered list.

File: guides/common/modules/proc_upgrading-host-packages.adoc [hunk 5]
```diff
@@ -0,0 +1,22 @@
+[id="Upgrading-Host-Packages_{context}"]
+= Upgrading Host Packages
+
+The package upgrade process takes place via Katello agent or remote execution, depending on your configuration.
+
+.Procedure
+. If the upgrade is available for a particular package, it is displayed in *Upgradable to*.
+. Upgrade them via *Remote Execution* or *Customized Remote Execution* from the vertical ellipsis on the right.
+
+To upgrade multiple packages at a time, select the *Upgradable* filter from the *Status* box to list the upgradable packages or directly select the packages that are upgradable.
```
Comment (maximiliankolb) [inline guides/common/modules/proc_upgrading-host-packages.adoc:10]: Also unsure if this is part of the procedure or not.

File: guides/common/modules/proc_managing-host-packages.adoc (hunk 2 above)
Comment (Lennonka) [reply guides/common/modules/proc_managing-host-packages.adoc:10]: I don't think we should include this.

**Comment by Lennonka** [comment]
These procedures are already covered in the following sections of _Managing Content_:
- https://docs.theforeman.org/nightly/Managing_Content/index-katello.html#Installing_Packages_on_a_Host_content-management
- https://docs.theforeman.org/nightly/Managing_Content/index-katello.html#upgrading-packages-on-a-host-in-bulk_content-management
- https://docs.theforeman.org/nightly/Managing_Content/index-katello.html#removing-packages-from-a-host_content-management

A couple of points:

1. Let's not duplicate them.
2. Perhaps you could just extend them if any workflows have changed. However, I would be careful when these procedures were written (before of after this particular demo), because they may actually be more up-to-date than the demo. I suggest you investigate the commits of the files vs. the date of the demo.
3. I'm wondering if they should be grouped into a single chapter, i.e. "Managing Packages"?

**Comment by mjivraja** [comment]
> These procedures are already covered in the following sections of _Managing Content_:
> 
> * https://docs.theforeman.org/nightly/Managing_Content/index-katello.html#Installing_Packages_on_a_Host_content-management
> * https://docs.theforeman.org/nightly/Managing_Content/index-katello.html#upgrading-packages-on-a-host-in-bulk_content-management
> * https://docs.theforeman.org/nightly/Managing_Content/index-katello.html#removing-packages-from-a-host_content-management
> 
> A couple of points:
> 
> 1. Let's not duplicate them.
> 2. Perhaps you could just extend them if any workflows have changed. However, I would be careful when these procedures were written (before of after this particular demo), because they may actually be more up-to-date than the demo. I suggest you investigate the commits of the files vs. the date of the demo.
> 3. I'm wondering if they should be grouped into a single chapter, i.e. "Managing Packages"?

Thanks, @Lennonka for this :smiley: 

1 & 2) I see that the procedures are the same (mine vs those you shared) and it may need some tweaks. The [demo](https://www.youtube.com/watch?v=ykQFM7aFczY&t=1471s) is old about 10 months while the file was last updated on 09/19/22. Ideally, the procedures in the files should be up-to-date but I feel it requires some refinement.

We avoided duplication here. :star: 

3) I can definitely merge sections 12, 13 & 14 into one i.e. `Managing Packages`. 

Need your call.

**Comment by Lennonka** [comment]
> 3. I can definitely merge sections 12, 13 & 14 into one i.e. `Managing Packages`.
You can go ahead.

**Review by Lennonka — CHANGES_REQUESTED** [review]
I'm sorry, Malhar, I realize that I merged another PR with this approach, but we shouldn't generally put multiple headings into a single file.
Ideally, there should be a single heading per module.
So, keeping the original files, add a concept module that will contain the chapter heading with the first paragraph and increase the leveloffset of the sections in `master.adoc`, like this:
```
include::common/modules/con_managing-packages-on-a-host.adoc[leveloffset=+1]

include::common/modules/proc_installing-packages-on-a-host.adoc[leveloffset=+2]

include::common/modules/proc_upgrading-packages-on-a-host-in-bulk.adoc[leveloffset=+2]

include::common/modules/proc_removing-packages-from-a-host.adoc[leveloffset=+2]
```

**Comment by mjivraja** [comment]
> 

No problem @Lennonka I have made the changes as per your comment. :+1: 

Just a silly question. Why can't we have an assembly for this?

**Comment by mjivraja** [comment]
> I'm sorry, Malhar, I realize that I merged another PR with this approach, but we shouldn't generally put multiple headings into a single file. Ideally, there should be a single heading per module. So, keeping the original files, add a concept module that will contain the chapter heading with the first paragraph and increase the leveloffset of the sections in `master.adoc`, like this:
> 
> ```
> include::common/modules/con_managing-packages-on-a-host.adoc[leveloffset=+1]
> 
> include::common/modules/proc_installing-packages-on-a-host.adoc[leveloffset=+2]
> 
> include::common/modules/proc_upgrading-packages-on-a-host-in-bulk.adoc[leveloffset=+2]
> 
> include::common/modules/proc_removing-packages-from-a-host.adoc[leveloffset=+2]
> ```

No problem @Lennonka  I have made the changes as per your comment. +1

Just a silly question. Why can't we have an assembly for this?

**Comment by mjivraja** [comment]
> I'm sorry, Malhar, I realize that I merged another PR with this approach, but we shouldn't generally put multiple headings into a single file. Ideally, there should be a single heading per module. So, keeping the original files, add a concept module that will contain the chapter heading with the first paragraph and increase the leveloffset of the sections in `master.adoc`, like this:
> 
> ```
> include::common/modules/con_managing-packages-on-a-host.adoc[leveloffset=+1]
> 
> include::common/modules/proc_installing-packages-on-a-host.adoc[leveloffset=+2]
> 
> include::common/modules/proc_upgrading-packages-on-a-host-in-bulk.adoc[leveloffset=+2]
> 
> include::common/modules/proc_removing-packages-from-a-host.adoc[leveloffset=+2]
> ```

No problem @Lennonka I have made the changes as per your comment. :+1: 

Just a silly question. Why can't we have an assembly for this?

**Comment by Lennonka** [comment]
> Why can't we have an assembly for this?

Good point, we can. You can add those modules into an assembly and include the assembly `master.adoc`.

**Comment by adamlazik1** [comment]
Oh no!
@mjivraja my apologies, I did not realize this PR was in progress when I created #2000 (I got it reported as bugzilla). Basically I created the assembly and moved it to **Managing Hosts**. Do you think we could somehow sync so the content isn't in two places? Perhaps you can apply these changes to the assembly that I moved?

**Comment by mjivraja** [comment]
> Oh no! @mjivraja my apologies, I did not realize this PR was in progress when I created #2000 (I got it reported as bugzilla). Basically I created the assembly and moved it to **Managing Hosts**. Do you think we could somehow sync so the content isn't in two places? Perhaps you can apply these changes to the assembly that I moved? (Although I only applied changes to 3.5)

Oww yes, I see you have merged for `3.5`. However, the content is a bit different in both PRs. 
I took many changes and made it nearly accurate with the help of Lena. Can we merge this content (this PR) on top of yours for all versions I marked? (Also, I resolved conflicts today)

**Comment by adamlazik1** [comment]
> Can we merge this content (this PR) on top of yours for all versions I marked? (Also, I resolved conflicts today).

We definitely can, although I thought managing packages was a thing only since 3.3. (at least with the new UI chages). If you have it confirmed that this is applicable all the way down to 3.1. then we can definitely merge that.

However, I think that assembly I created is not necessary with changes you are making in this PR. So what I was thinking is  whether you move your assembly to **Managing Hosts** (as I think the content is more related to managing hosts than managing content because all the procedures are applied to individual hosts). The situation is a bit different on 3.5. So I would kindly ask you to make another commit that will only get cherry-picked to 3.5, where:
* You delete assembly I created in #2000 as it is no longer necessary with your more actualized version
* You include module **Enabling and Disabling repositories on hosts** in your assembly, which I originally put in my assembly as it very closely relates to managing packages (originaly this module was included in **Managing Activation Keys** which did not make a whole lotta sense so I had to move it). If you want you can rename the assembly on 3.5 to **Managing Repositories and Packages on Hosts** to make it more accurate.

Again I do apologize for making a mess but I think it is not necessary to duplicate content so I would prefer only having one assembly and in one place.
Another option would be that I can make the aforementioned commit in a different PR once this gets merged.

What do you think?

**Comment by mjivraja** [comment]
> > Can we merge this content (this PR) on top of yours for all versions I marked? (Also, I resolved conflicts today).
> 
> We definitely can, although I thought managing packages was a thing only since 3.3. (at least with the new UI chages). If you have it confirmed that this is applicable all the way down to 3.1. then we can definitely merge that.
> 
> However, I think that assembly I created is not necessary with changes you are making in this PR. So what I was thinking is whether you could move your assembly to **Managing Hosts** (as I think the content is more related to managing hosts than managing content because all the procedures are applied to individual hosts). The situation is a bit different on 3.5. So I would kindly ask you to make another commit that will only get cherry-picked to 3.5, where:
> 
> * You delete assembly I created in [Move host-related modules to Managing Hosts #2000](https://github.com/theforeman/foreman-documentation/pull/2000) as it is no longer necessary with your more actualized version
> * You include module **Enabling and Disabling repositories on hosts** in your assembly, which I originally put in my assembly as it very closely relates to managing packages (originaly this module was included in **Managing Activation Keys** which did not make a whole lotta sense so I had to move it). If you want you can rename the assembly on 3.5 to **Managing Repositories and Packages on Hosts** to make it more accurate.
> 
> Again I do apologize for making a mess but I think it is not necessary to duplicate content so I would prefer only having one assembly and in one place. Another option would be that I can make the aforementioned commit in a different PR once this gets merged.
> 
> What do you think?

I read all your points and views. Let's make it straight-forward!

If we merge this PR, it will be even easy from your side to make further modifications applicable to **Managing Hosts** and the new assembly you created. You can definitely move content best to your knowledge and simultaneously this PR will be clean from the reviewers' point because it will focus on a single thing at a time.

**Comment by adamlazik1** [comment]
Understood! I will make my changes in a different PR once this gets merged. Leaving rest of the review to @Lennonka.

File: guides/common/modules/con_managing-packages-on-a-host.adoc [hunk 6]
```diff
@@ -0,0 +1,5 @@
+[id="Managing_Packages_on_a_Host_{context}"]
+= Managing Packages on a Host
+
+On the {ProjectWebUI}, you can have a comprehensive package management system to install, upgrade, and remove packages on a host.
+You can manage the packages of hosts even if the subscriptions are not attached to access repositories.
```
Comment (Lennonka) [inline guides/common/modules/con_managing-packages-on-a-host.adoc:5]: I'm not sure about this sentence, I would avoid this claim altogether.

File: guides/common/modules/proc_installing-packages-on-a-host.adoc [hunk 7]
```diff
@@ -7,7 +7,7 @@ The list of packages available for installation depends on the Content View and
 .Procedure
 . In the {ProjectWebUI}, navigate to *Hosts* > *All Hosts* and select the host you want to install packages on.
 . On the *Content* tab, select the *Packages* tab.
-. On the vertical ellipsis icon next to the upgrade button, click *Install Packages*.
+. On the vertical ellipsis icon at the top, click *Install Packages*.
```
Comment (Lennonka) [inline guides/common/modules/proc_installing-packages-on-a-host.adoc:10]: We should avoid this kind of navigation as per the IBM Style Guide.

File: guides/common/modules/proc_upgrading-packages-on-a-host.adoc [hunk 8]
```diff
@@ -1,14 +1,28 @@
...
 . In the {ProjectWebUI}, navigate to *Hosts* > *All Hosts*.
 . Click the name of the host you want to modify.
 . Click the *Content* tab, then click the *Packages* tab.
 . Select *Upgradable* from the *Status* list.
-. Select upgrade version from the dropdown menu in *Upgradable to* column where applicable.
+. Select an upgrade version from the dropdown menu in *Upgradable to* column where applicable.
 . Select the packages you want to upgrade.
-. Click the *Upgrade* button.
+. Click *Remote Execution* or *Customized Remote Execution* from the vertical ellipsis on the right of the package.
+** To upgrade multiple packages at a time, select the *Upgradable* filter from the *Status* box to list the upgradable packages.
```
Comment (Lennonka) [inline guides/common/modules/proc_upgrading-packages-on-a-host.adoc:15]: This line is duplicating information that is already in the steps 4. and 5.

File: guides/common/modules/proc_upgrading-packages-on-a-host.adoc [hunk 9]
```diff
@@ -1,14 +1,28 @@
...
+
+You can run a remote execution job immediately, or select to customize the remote execution job.
+
+[NOTE]
+====
+If a package has more than one available upgrade version, only the latest upgradeable version is displayed.
```
Comment (Lennonka) [inline guides/common/modules/proc_upgrading-packages-on-a-host.adoc:21]: This sentence might make sense in step 5. Adding it as a note below the procedure is out of context.

File: guides/common/modules/proc_upgrading-packages-on-a-host.adoc [hunk 10]
```diff
@@ -1,14 +1,28 @@
...
+ifdef::katello[]
+This behavior has been replaced by a dropdown list of all updates.
```
Comment (Lennonka) [inline guides/common/modules/proc_upgrading-packages-on-a-host.adoc:23]: The "dropdown list" is already mentioned as the "dropdown menu" in the step 5.

File: guides/common/modules/proc_upgrading-packages-on-a-host.adoc [hunk 11]
```diff
@@ -1,14 +1,28 @@
...
+Please check the Katello version when it is changed.
```
Comment (Lennonka) [inline guides/common/modules/proc_upgrading-packages-on-a-host.adoc:24]: Are you seriously telling the user to check for themself when the feature was added? It is our responsibility to describe features correctly in each version of the docs.

Additionally, the Katello version always corresponds with a particular Foreman version as you can see in the [PR template](https://github.com/theforeman/foreman-documentation/blob/master/.github/PULL_REQUEST_TEMPLATE.md).

File: guides/common/modules/proc_removing-packages-from-a-host.adoc [hunk 12]
```diff
@@ -1,11 +1,13 @@
...
+You can remove a single package or multiple packages from a host in the {ProjectWebUI}.
 
 .Procedure
 . In the {ProjectWebUI}, navigate to *Hosts* > *All Hosts*.
 . Click the name of the host you want to modify.
 . Click the *Content* tab, then click the *Packages* tab.
 . Click the vertical ellipsis for the package you want to remove and select *Remove*.
-You get a REX job notification once the remote execution job is complete.
\ No newline at end of file
+** To remove multiple packages, select the packages by checking the boxes and clicking *Remove* from the vertical ellipsis at the top.
```
Comment (Lennonka) [inline guides/common/modules/proc_removing-packages-from-a-host.adoc:11]: If you want to make the workflow for multiple packages, it should be rewritten properly, i.e. it should look more like the original Upgrading packages procedure.

**Review by Lennonka — COMMENTED** [review]
I know that you made your suggestions in good faith, but I'm for closing this PR (and your ticket) as already documented.
Your suggestions actually make the procedures more confusing, you duplicate information and place it out of context. (I tried to mention a couple of concrete explanations below, but it's not all of the issues.)
Have you tried to perform the procedures in the GUI for yourself while working on them?

**Comment by mjivraja** [comment]
> I know that you made your suggestions in good faith, but I'm for closing this PR (and your ticket) as already documented. Your suggestions actually make the procedures more confusing, you duplicate information and place it out of context. (I tried to mention a couple of concrete explanations below, but it's not all of the issues.)

I drafted the content based on the Foreman Demo (link in the description of this PR) and submitted it for SME review. I received some suggestions from Lucy Fu which were technical and applied them. 

I can see that some changes do not look good to you and I am ready to revert. Also, I will re-work on content based on your observation and not limited to your suggestions.

> Have you tried to perform the procedures in the GUI for yourself while working on them?

Yes. I drafted the content by watching the video. The content may not adhere to the style guide but I verified it on the instance as well.

File: guides/common/modules/proc_installing-packages-on-a-host.adoc (hunk 7 above)
Comment (mjivraja) [reply guides/common/modules/proc_installing-packages-on-a-host.adoc]: Addressed :+1:

File: guides/common/modules/proc_upgrading-packages-on-a-host.adoc (hunk 8 above)
Comment (mjivraja) [reply guides/common/modules/proc_upgrading-packages-on-a-host.adoc]: Addressed :+1:

File: guides/common/modules/proc_upgrading-packages-on-a-host.adoc (hunk 9 above)
Comment (mjivraja) [reply guides/common/modules/proc_upgrading-packages-on-a-host.adoc]: Addressed :+1:

File: guides/common/modules/proc_upgrading-packages-on-a-host.adoc (hunk 10 above)
Comment (mjivraja) [reply guides/common/modules/proc_upgrading-packages-on-a-host.adoc]: Addressed :+1:

File: guides/common/modules/proc_upgrading-packages-on-a-host.adoc (hunk 11 above)
Comment (mjivraja) [reply guides/common/modules/proc_upgrading-packages-on-a-host.adoc]: Removed the note.

File: guides/common/modules/proc_removing-packages-from-a-host.adoc (hunk 12 above)
Comment (mjivraja) [reply guides/common/modules/proc_removing-packages-from-a-host.adoc]: Completely changed the workflow in both. :+1:

**Comment by mjivraja** [comment]
When I originally drafted this, the version of instance I received was older. I have now updated the content based on the `6.13` instance. Also, I have tried to keep the procedure even straigh-forward.

Apologise for the inconvenience @Lennonka

File: guides/common/modules/proc_upgrading-packages-on-a-host.adoc [hunk 13]
```diff
@@ -1,14 +1,18 @@
...
 . Click the *Content* tab, then click the *Packages* tab.
-. Select *Upgradable* from the *Status* list.
-. Select upgrade version from the dropdown menu in *Upgradable to* column where applicable.
-. Select the packages you want to upgrade.
-. Click the *Upgrade* button.
+. Select *Upgradable* from the *Status* dropdown menu.
+. Select the packages from list and click *Upgrade*.
+If a package has more than one available upgrade version, only the latest upgradable version is displayed.
+
+You can also run a remote execution job immediately by selecting *Upgrade via remote execution* from the *Upgrade* dropdown menu.
```
Comment (mjivraja) [inline guides/common/modules/proc_upgrading-packages-on-a-host.adoc:15]: We can also add the link to `Configuring and Setting Up the Remote Execution`, same as the one described in Installing Package part (last lines).

**Comment by Lennonka** [comment]
I think we should rebase the changes on Adam's changes to avoid an unnecessary duplication. It's easy:
- rebase on master
- delete your `con_*` file
- delete your assembly and remove it from the master.adoc in Managing Content
- you should still be able to see the modifications in modules

**Comment by mjivraja** [comment]
> I think we should rebase the changes on Adam's changes to avoid an unnecessary duplication. It's easy:
> 
> * rebase on master
> * delete your `con_*` file
> * delete your assembly and remove it from the master.adoc in Managing Content
> * you should still be able to see the modifications in modules
> 
> If your changes are supposed to be just for 6.13, then we shouldn't cherry-pick them to 3.4 and lower, and we can simply rebase.

My changes were supposed to be applicable to `6.11` and later. However, Adam also worked on the same cause. I don't know if his PR for the same got merged. As per our talk [here](https://github.com/theforeman/foreman-documentation/pull/1893#issuecomment-1434624882) we agreed to merge this PR first.

But, if we are saving time. we can totally close this PR considering Adam's PR is enough and can be cherry-picked up to `6.11`. It will still fulfill the requirement of my ticket :+1:

**Comment by adamlazik1** [comment]
I suppose it could be cherry-picked but would require some manual  edits since it contains module only accessible from 3.5.

**Comment by adamlazik1** [comment]
Yep, that's the one.

**Comment by mjivraja** [comment]
Alright, we can move forward to close this if it's clear. 👍

**Comment by adamlazik1** [comment]
You can also just use this PR and this branch to cherry-pick changes from #2000 to lower branches if you wish to do so. However; I would like to turn your attention to this: files included in the new assembly only exist since 3.3, that is why I assume this also only applies to 3.3 and onwards. There is also the fact that under the video you linked was released when Foreman 3.2 was released, which further enhances this idea.

If you wish to cherry-pick #2000 to 3.4 and 3.3, you can perform when following:
1. Reset the branch to 3.4. This will delete all changes you made in this PR so make sure to create a backup branch if you want to keep some for further modification.
  ```
  git reset --hard 3.4
  ```
2. cherry-pick c610080 to create assembly in Managing Hosts
3. Remove `include::modules/proc_enabling-and-disabling-repositories-on-hosts.adoc[leveloffset=+1]` from the assembly since the module is not accessible until 3.5
4. Change assembly description to `You can use {Project} to install, upgrade, and remove packages on hosts.` (Removing part about toggling repositories)
5. Include previous changes in the cherry-picked commit
6. cherry-pick 183b731 to change Upgrading packages module name to what it currently is on 3.5
7. Change the target branch of the PR to 3.4 and select a cherry-pick to 3.3.

You can choose to keep those two commits separate as I did or you can choose to squash them into one commit.

**Comment by mjivraja** [comment]
> You can also just use this PR and this branch to cherry-pick changes from #2000 to lower branches if you wish to do so. However; I would like to turn your attention to this: files included in the new assembly only exist since 3.3, that is why I assume this also only applies to 3.3 and onwards. There is also the fact that under the video you linked was released when Foreman 3.2 was released, which further enhances this idea.
> 
> If you wish to cherry-pick #2000 to 3.4 and 3.3, you can perform when following:
> 
> 1. Reset the branch to 3.4. This will delete all changes you made in this PR so make sure to create a backup branch if you want to keep some for further modification.
> 
> ```
> git reset --hard 3.4
> ```
> 
> 2. cherry-pick [c610080](https://github.com/theforeman/foreman-documentation/commit/c6100808d141f26f7492f0590aa65b86021ddc11) to create assembly in Managing Hosts
> 3. Remove `include::modules/proc_enabling-and-disabling-repositories-on-hosts.adoc[leveloffset=+1]` from the assembly since the module is not accessible until 3.5
> 4. Change assembly description to `You can use {Project} to install, upgrade, and remove packages on hosts.` (Removing part about toggling repositories)
> 5. Include previous changes in the cherry-picked commit
> 6. cherry-pick [183b731](https://github.com/theforeman/foreman-documentation/commit/183b731591b5a0208c66fa430e437efd7f7fe341) to change Upgrading packages module name to what it currently is on 3.5
> 7. Change the target branch of the PR to 3.4 and select a cherry-pick to 3.3.
> 
> You can choose to keep those two commits separate as I did or you can choose to squash them into one commit.

Is there any other idea/process that is less complex? I fear we may mess something up.

**Comment by adamlazik1** [comment]
Well, instead of cherry-picking you can manually apply the changes but I think this should be straightforward. Feel free to contact me in private so we can discuss this in more detail.

**Review by adamlazik1 — APPROVED** [review]

**Comment by adamlazik1** [comment]
Merged and cherry-picked:
  919886859..854e6f8ba  3.3 -> 3.3

Resolved a small merge conflict.
